Gems/Terrain/Code/Tests/TerrainSystemTest.cpp

```diff
@@ -226,3 +267,202 @@ TEST_F(TerrainSystemTest, TerrainExistsOnlyWithinTerrainLayerSpawnerBounds)
+        AZ::Vector3(-5.5f, -5.5f, -11.0f), // Should return a height of -5.50 + -5.50
+        AZ::Vector3(-7.7f, -7.7f, -15.5f)  // Should return a height of -7.75 + -7.75
+    };
+    for (auto& testPoint : testPoints)
+    {
+        const float expectedHeight = testPoint.GetZ();
+
+        AZ::Vector3 position(testPoint.GetX(), testPoint.GetY(), 0.0f);
+        bool heightQueryTerrainExists = false;
+        float height =
+            m_terrainSystem->GetHeight(position, AzFramework::Terrain::TerrainDataRequests::Sampler::CLAMP, &heightQueryTerrainExists);
+
+        constexpr float epsilon = 0.0001f;
+        EXPECT_NEAR(height, expectedHeight, epsilon);
+    }
+}
+
+TEST_F(TerrainSystemTest, TerrainHeightQueriesWithBilinearSamplersUseQueryGridToInterpolate)
+{
+    // Verify that when using the "BILINEAR" height sampler, the heights are interpolated from points sampled from the query grid.
+
+    // Create a mock terrain layer spawner that uses a box of (-10,-10,-5) - (10,10,15) and generates a height equal
+    // to the X + Y position, so we'll have heights that look like this on our grid:
+    //   0 *---* 1
+    //     |   |
+    //   1 *---* 2
+    // However, everywhere inside the grid box, we'll generate heights much larger than X + Y.  It will have no effect on exact grid
+    // points, but it will noticeably affect the expected height values if any points get sampled in-between grid points.
+
+    const AZ::Aabb spawnerBox = AZ::Aabb::CreateFromMinMaxValues(-10.0f, -10.0f, -5.0f, 10.0f, 10.0f, 15.0f);
+    const float amplitudeMeters = 10.0f;
+    const float frequencyMeters = 1.0f;
+    auto entity = CreateAndActivateMockTerrainLayerSpawner(
+        spawnerBox,
+        [amplitudeMeters, frequencyMeters](AZ::Vector3& position, bool& terrainExists)
+        {
+            // Our generated height will be X + Y.
+            float expectedHeight = position.GetX() + position.GetY();
+
+            // If either X or Y aren't evenly divisible by the query frequency, add a scaled value to our generated height.
+            // This will show up as an unexpected height "spike" if it gets used in any bilinear filter queries.
+            float unexpectedVariance = amplitudeMeters *
+                (fmodf(position.GetX(), frequencyMeters) + fmodf(position.GetY(), frequencyMeters));
+            position.SetZ(expectedHeight + unexpectedVariance);
+            terrainExists = true;
+        });
+
+    // Create and activate the terrain system with our testing defaults for world bounds, and a query resolution at 1 meter intervals.
+    const AZ::Vector2 queryResolution(frequencyMeters);
+    CreateAndActivateTerrainSystem(queryResolution);
+
+    // Test some points and verify that the results are the expected bilinear filtered result,
+    // whether they're in positive or negative space.
+    // (Z contains the the expected result for convenience).
+    const AZ::Vector3 testPoints[] = {
```

Comment:
Makes sense, I'll change it!  The Vector3 made sense to me because the point represented the full "expected value" (the z at this x and y), but if the struct is more readable to others, I'm totally good with that.  The struct will also make more sense when writing similar types of tests for normals and other data, so it's definitely more flexible going forward.